tools/devops/automation/templates/common/checkout.yml

```
@@ -22,29 +22,10 @@ steps:
     submodules: recursive
     path: s/xamarin-macios
 
-- ${{ if parameters.isPR }}:
-  - pwsh: |
-      git config remote.origin.fetch '+refs/pull/*:refs/remotes/origin/pull/*'
-      git fetch origin
-      $branch="$(Build.SourceBranch)".Replace("merge", "head")
-      $branch=$branch.Replace("refs", "origin")
-      Write-Host "Checking out branch $branch"
-      git checkout $branch
-      git submodule update --recursive --init
-      git clean -xffd
-      git submodule foreach --recursive git clean -xffd
-      $hash = git rev-parse HEAD
-      Write-Host "##vso[task.setvariable variable=GIT_HASH;isOutput=true]$hash"
-    name: fix_commit
-    displayName: "Undo Github merge"
-    workingDirectory: $(System.DefaultWorkingDirectory)/xamarin-macios
-- ${{ else }}:
-  - pwsh: |
-      $hash = git rev-parse HEAD
-      Write-Host "##vso[task.setvariable variable=GIT_HASH;isOutput=true]$hash"
-    name: fix_commit
-    displayName: "Undo Github merge"
-    workingDirectory: $(System.DefaultWorkingDirectory)/xamarin-macios
+- pwsh: undo_github_merge.ps1 -SourceBranch $(Build.SourceBranch) -IsPr "${{ parameters.isPR }}"
```

Still a yaml novice but this is where pipeline is failing right now: 
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=7617125&view=logs&j=443a5880-1958-5052-6859-a80de6521633&t=cf3ae589-2ffd-5270-362c-14dea36c810c&l=12

Everything else looks good, so I believe this should fix it?
```suggestion
- pwsh: undo_github_merge.ps1 -SourceBranch "$(Build.SourceBranch)" -IsPr "${{ parameters.isPR }}"
```

---

good catch, has been a fun day, thanks for the help :)